proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,751 @@
+syntax = "proto3";
+
+package redpanda.api.console.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+message Security {
+  enum PrincipalType {
+    PRINCIPAL_TYPE_UNSPECIFIED = 0;
+    PRINCIPAL_TYPE_USER = 1;
+    PRINCIPAL_TYPE_GROUP = 2;
+    PRINCIPAL_TYPE_REDPANDA_ROLE = 3;
+  }
+}
+
+// Role defienes a role in the system.
+message Role {
+  // The name of the role.
+  string name = 1;
+}
```

Comment:
Wondering whether this type is actually worth it. As of today it's just one property and we have some responses (e.g. `GetRoleResponse`) where we just return a string instead of a `Role`.

Reply:
Hmm I am not sure if RFC got updated, or perhaps my definition here was incorrect from start, but `GetRoleResponse` should return the role and all members. I have updated the type to reflect that.
Although we don't adhere to it strictly perhaps, it's considered best practice for requests and responses not to have primitive types, except for ids and tokens, but to contain the "resource object" as a field. I have changed the API to reflect that.